tests/integration/test_arg_reduce.py

```diff
@@ -20,26 +20,145 @@
 from legate.core import LEGATE_MAX_DIM
 
 
-@pytest.mark.parametrize("ndim", range(LEGATE_MAX_DIM + 1))
-@pytest.mark.parametrize("keepdims", [True, False])
-def test_argmax_and_argmin(ndim, keepdims):
-    shape = (5,) * ndim
-
-    in_np = np.random.random(shape)
-    in_num = num.array(in_np)
-
-    for fn in ("argmax", "argmin"):
-        fn_np = getattr(np, fn)
-        fn_num = getattr(num, fn)
-        assert np.array_equal(
-            fn_np(in_np, keepdims=keepdims), fn_num(in_num, keepdims=keepdims)
-        )
-        if in_num.ndim == 1:
-            continue
-        for axis in range(in_num.ndim):
-            out_np = fn_np(in_np, axis=axis, keepdims=keepdims)
-            out_num = fn_num(in_num, axis=axis, keepdims=keepdims)
-            assert np.array_equal(out_np, out_num)
+class TestArgReduceErrors:
+    """
+    this class is to test negative cases
+    argmax(a, axis=None, out=None, *, keepdims=np._NoValue):
+    argmin(a, axis=None, out=None, *, keepdims=np._NoValue)
+    """
+
+    def test_empty_ary(self):
+        # empty array should not be accepted
+        msg = r"an empty sequence"
+        for fn in ("argmax", "argmin"):
+            fn_num = getattr(num, fn)
+            with pytest.raises(ValueError, match=msg):
+                fn_num([])
```

Comment:
```suggestion
    @pytest.mark.parametrize("fn", ("argmax", "argmin"))
    def test_empty_ary(self, func_name):
        msg = r"an empty sequence"
        func = getattr(num, func_name)
        with pytest.raises(ValueError, match=msg):
            func([])
```
Generally speaking, test should avoid loops inside. If there are multiple cases to handle, that should usually be done with `pytest.mark.parametrize` so that there is better test resolution.

Reply:
Fix it in new commit.